src/lib/operators/projection.cpp

```diff
@@ -216,9 +216,17 @@ std::shared_ptr<const Table> Projection::_on_execute() {
 
     // The output chunk contains all rows that are in the stored chunk, including invalid rows. We forward this
     // information so that following operators (currently, the Validate operator) can use it for optimizations.
-    output_chunks[chunk_id] =
-        std::make_shared<Chunk>(std::move(output_chunk_segments[chunk_id]), input_chunk->mvcc_data());
-    output_chunks[chunk_id]->increase_invalid_row_count(input_chunk->invalid_row_count());
+    const auto chunk = std::make_shared<Chunk>(std::move(output_chunk_segments[chunk_id]), input_chunk->mvcc_data());
+    chunk->increase_invalid_row_count(input_chunk->invalid_row_count());
+
+    // Set ordered_by flag
+    const auto ordered_by = input_chunk->ordered_by();
+    if (ordered_by) {
+      chunk->finalize();
```

Comment:
Yes, good point. It could indeed be misleading to couple `ordered_by` and `finalize`. But then we have the same situation that you mentioned for `sort.cpp`, right?

>  When would chunk not be mutable? Could this be an assert?

Reply:
Not sure if I can follow. In `sort.cpp`, you checked for the mutable flag. My point over there was that the check seems redundant. We created the chunk, so it should be mutable. If it is already flagged as immutable, we are in an unexpected state, which should be reported, not ignored.

Reply:
Changed in 37674125cee4db2b782d719cc28f3e3b43cc8854